site/blog/unlisted/subqueries.md

```diff
@@ -0,0 +1,48 @@
+---
+slug: subquery-vs-join
+title: Subquery vs JOIN
+authors: [depombo]
+tags: [sql]
+---
+
+As a developer, you are probably familiar with the concept of joins in SQL databases. Joins are a powerful tool for combining data from multiple tables, but there are certain situations where subqueries may be a better choice. In this blog post, we will look at the advantages of using subqueries instead of joins in PostgreSQL, and why it may be the better option for your project.
+
+### Improved Performance
+One of the biggest advantages of using subqueries is improved performance. Subqueries can be optimized and executed more efficiently than joins, especially when dealing with large amounts of data. In PostgreSQL, subqueries can be executed in parallel, which can significantly improve performance. By breaking down the query into smaller parts, the database can use its indexing and caching mechanisms to optimize the performance of the query. In contrast, joins can be slower and more resource-intensive, especially when dealing with complex queries or large datasets. 
+
+### Better Readability
+Another advantage of using subqueries is improved readability. Subqueries can make your SQL code cleaner and easier to understand, especially for developers who are new to your project. Joins can be difficult to follow, especially when there are many tables involved or when the relationships between tables are complex. Subqueries can help simplify the code and make it easier for others to follow or reuse your logic.
+
+### Increased Flexibility
+Subqueries also offer increased flexibility compared to joins. With subqueries, you have more control over the order in which the data is retrieved, and you can easily modify the query to return the data you need. Joins are often more rigid, and making changes to the query can be more complex and time-consuming. In PostgreSQL, subqueries can be used to retrieve data from a single table and then join the results with another table, eliminating the need for complex join conditions.
+
+### Simplified Maintenance
```

Comment:
Would be good to explicitly call out composability here. You could literally have the different sub queries defined in other source files and included in the top-level query by templating. Improvements or schema changes that affect one sub query then are encapsulated away from the logic of the high-level query.

For instance you can call that out in the example below by introducing a new column on the department that affects the WHERE clause. Maybe 'Engineering' is split between 'Software' and 'Hardware Development' (so a simple `LIKE %Engineering` won't work) but they both are joined to `cost_center` table under the `R&D` category that could use a third sub-query?

Reply:
Implemented!